native/packages/react-native-bpk-component-progress/src/BpkProgress.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  StyleSheet,
+  View,
+  Animated,
+  Easing,
+  Platform,
+  ViewPropTypes,
+} from 'react-native';
+import {
+  animmationDurationBase,
+  spacingMd,
+  colorGray100,
+  colorGreen500,
+  colorGreen50,
+  borderRadiusPill,
+  spacingXxl,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  container: {
+    height: spacingMd,
+    overflow: 'hidden',
+  },
+  fill: {
+    height: spacingMd,
+  },
+  defaultTrackStyle: {
+    backgroundColor: Platform.select({
+      ios: () => colorGray100,
+      android: () => colorGreen50,
+    })(),
+    borderRadius: borderRadiusPill,
+    width: spacingXxl * 3,
+  },
+  defaultFillStyle: {
+    backgroundColor: colorGreen500,
+    borderRadius: borderRadiusPill,
+  },
+  barTrackStyle: {
+    backgroundColor: Platform.select({
+      ios: () => colorGray100,
+      android: () => colorGreen50,
+    })(),
+    width: '100%',
+  },
+  barFillStyle: {
+    backgroundColor: colorGreen500,
+  },
+});
+
+const BAR_TYPES = {
+  default: 'default',
+  bar: 'bar',
+};
+
+type Props = {
+  max: number,
+  min: number,
+  value: number,
+  type: $Keys<typeof BAR_TYPES>,
+  style: ?any,
+  fillStyle: ?any,
+};
+
+type State = {
+  width: number,
+};
+
+class ProgressBar extends React.Component<Props, State> {
+  progressAnimation: Animated.Value;
+
+  static defaultProps = {
+    max: 100,
+    min: 0,
+    type: BAR_TYPES.default,
+    style: null,
+    fillStyle: null,
+  };
+
+  static propTypes = {
+    max: PropTypes.number,
+    min: PropTypes.number,
+    value: PropTypes.number.isRequired,
+    type: PropTypes.oneOf(Object.keys(BAR_TYPES)),
+    style: ViewPropTypes.style,
+    fillStyle: ViewPropTypes.style,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.progressAnimation = new Animated.Value(
+      Math.min(props.value, props.max),
+    );
+    this.state = {
+      width: 0,
+    };
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.value >= 0 && this.props.value !== prevProps.value) {
+      Animated.timing(this.progressAnimation, {
+        easing: Easing.inOut(Easing.ease),
+        duration: animmationDurationBase,
+        toValue: Math.min(this.props.value, this.props.max),
```

Comment:
We can do that too but I don't see any use case for it. Limiting the max is to make sure the transition works as expected, because setting a value greater than the max would cause it to move a bit faster. I don't see anyone using a progress starting from the max and going down.

Reply:
You never know and better safe than sorry I guess :D

Reply:
I feel we are adding code to support something that's not meant to be supported but it's simple enough so :shrug:

Reply:
It's not unforseeable that progress could go backwards. If you add more work to the total of the progress it will go backwards for example

Reply:
Sure, I just don't think it will go bellow the min, and this is already taken care of by the `interpolate` function bellow, but I've made the changes anyways so I won't have to be sorry :D